File: src/main/java/rag/data/ChunkStore.java
package main.java.rag.data;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import main.java.rag.model.Chunk;


public class ChunkStore {
    private final Map<String, Chunk> chunks; // key: docId||chunkId
    private final Map<String, String> documentTitles; // key: docId, value: title
    
    public ChunkStore() {
        this.chunks = new HashMap<>();
        this.documentTitles = new HashMap<>();
    }
    
    
    public void addChunk(Chunk chunk) {
        String key = chunk.getDocId() + "||" + chunk.getChunkId();
        chunks.put(key, chunk);
    }
    
    
    public Chunk getChunk(String docId, String chunkId) {
        String key = docId + "||" + chunkId;
        return chunks.get(key);
    }
    
    
    public List<Chunk> getAllChunks() {
        return new ArrayList<>(chunks.values());
    }
    
    
    public void setDocumentTitle(String docId, String title) {
        documentTitles.put(docId, title);
    }
    
    
    public String getDocumentTitle(String docId) {
        return documentTitles.get(docId);
    }
    
    
    public Set<String> getAllDocIds() {
        return documentTitles.keySet();
    }
    
    
    public int size() {
        return chunks.size();
    }
}

File: iteration2/src/data/chunk_store.py
from typing import Dict, List, Set, Optional
from src.model.chunk import Chunk

class ChunkStore:
    
    def __init__(self) -> None:
        self.__chunks: Dict[str, Chunk] = {}  # key: "docId||chunkId"
        self.__document_titles: Dict[str, str] = {}  # key: docId, value: title

    def add_chunk(self, chunk: Chunk) -> None:
        key = f"{chunk.get_doc_id()}||{chunk.get_chunk_id()}"
        self.__chunks[key] = chunk

    def get_chunk(self, doc_id: str, chunk_id: str) -> Optional[Chunk]:
        key = f"{doc_id}||{chunk_id}"
        return self.__chunks.get(key)

    def get_all_chunks(self) -> List[Chunk]:
        return list(self.__chunks.values())

    def set_document_title(self, doc_id: str, title: str) -> None:
        self.__document_titles[doc_id] = title

    def get_document_title(self, doc_id: str) -> Optional[str]:
        return self.__document_titles.get(doc_id)

    def get_all_doc_ids(self) -> Set[str]:
        return set(self.__document_titles.keys())

    def size(self) -> int:
        return len(self.__chunks)
